### ISSUE TITLE
About generative modifier metrics

### ISSUE THREAD

**Issue by minjung98** [naver/posescript#24]
Hello. Thank you for your consistently quick and kind responses.
I have a question regarding the metrics results I obtained when reproducing the generate_modifier code. 
Specifically, I'm concerned about the fact that the metrics after fine-tuning are not better than those obtained from pretraining alone.
Without changing anything in the code, I trained for 3000 epochs during pretraining and 2000 epochs during fine-tuning. 
The results I obtained are as follows. 
**pretrain**
![스크린샷 2024-07-21 오후 6 08 49](https://github.com/user-attachments/assets/fd3a135c-456c-4972-b984-cae01ad47738)
**fine-tune**
![스크린샷 2024-07-21 오후 6 09 52](https://github.com/user-attachments/assets/f2e50d1c-878e-4192-983a-1bdab7908321)
I'd like to ask if you obtained similar results when you trained the model, and I'd also appreciate your insights on why the results turned out this way.

**Comment by g-delmas** [comment]
Hello!
Are you:
* pretraining on PoseFix-automatic
* evaluating the pretraining on PoseFix-automatic
* finetuning on PoseFix-human
* evaluating the finetuning on PoseFix-human
... or are you doing something different?

**Comment by minjung98** [comment]
I've done pretraining on PoseFix-automatic for **pretrain** and finetuning on PoseFix-human for **fine-tune**.
Pictures I added on question is the result of last validation metrics from this code. 
` metrics = compute_eval_metrics(unwrapped_model, dataset_val, self.device,
                                        pose_model=pose_model,
                                        fid_version=fid_version,
                                        textret_model=textret_model,
                                        tokenizer_name_textret_model=tokenizer_name_textret_model)`
It was on train_generative_modifier.py!

Thank you.

**Comment by g-delmas** [comment]
Then it makes sense that you are not getting "better" results; as they are not directly comparable. The first table was obtained on _PoseFix-A_, while the second table was obtained on _PoseFix-H._ These are basically two different datasets: PoseFix-A is larger, and contains automatic annotations, while PoseFix-H is smaller and contains human-written annotations. If you evaluate your pretrained model on PoseFix-H directly, you should find that the finetuned version is performing much better! (because it has adapted to real human-written formulations)

**Comment by minjung98** [comment]
Your observation is correct. The tables I showed you summarize the final results obtained after running all epochs for each respective phase(pretrain, finetune). So for the posefix-H dataset, I ran the following commands for evaluation:

`bash generative_modifier/script_generative_modifier.sh 'eval' /home/chanhun_posefix/chanhun/posescript/experiments/FeedbackGenerator_textdecoder-transformer_vocPFAHPP_tirg_dlatentD512_dnhead8_latentD32/train-posefix-A/crossentropy/B128_Adam_lr0.0001_wd0.0001/seed1/checkpoint_best.pth`

`bash generative_modifier/script_generative_modifier.sh 'eval' /home/chanhun_posefix/chanhun/posescript/experiments/FeedbackGenerator_textdecoder-transformer_vocPFAHPP_tirg_dlatentD512_dnhead8_latentD32/train-posefix-H/crossentropy/B128_Adam_lr1e-05_wd0.0001_LRflip_pretrained_modgen_CAtransfPFAHPP_dataPFAftPFH/seed1/checkpoint_best.pth`

The results were as follows: 
![image](https://github.com/user-attachments/assets/57ecd15a-51b5-42fe-9425-5494b9cc2e24)
![image](https://github.com/user-attachments/assets/2e3f6fdd-e0fd-4e09-bf6a-83b186364c80)

I have two questions about the result. 
1) Compared to the pretrained model, the model that was finetuned after pretraining showed better performance in text-related results. However, it seems to have performed worse in terms of r-precision. How similar are these results to what you experienced in your experiments?
2) From what I understand, it seems that the dataset used for training during the finetuning stage is the same as the dataset used for evaluation. Is my understanding correct? I'm curious if this fact might affect the evaluation results.

Thank you.

**Comment by minjung98** [comment]
I have an additional question: I would appreciate if you could explain what GT, random, and auto_posefix-A_cap0 are comparing in evaluate_generative_modifier. Thank you for your valuable time.

**Comment by g-delmas** [comment]
Hello,

Sorry for the late answer, this last month has been a bit hectic.

First thing to note is that the results you obtain on PoseFix-H after finetuning are similar to those I report on [this page](https://github.com/naver/posescript/blob/main/pretrained_models.md), which is a good sign.

1. What model are you using [here](https://github.com/naver/posescript/blob/f5282b1115ab09a077ba50a04b230a939306c604/src/text2pose/generative_modifier/script_generative_modifier.sh#L33C1-L33C14)? Is it the same for both experiments? I am using `modret_distilbert_dataPFAftPFH` which is [shared here](https://github.com/naver/posescript/blob/f5282b1115ab09a077ba50a04b230a939306c604/pretrained_models.md).
Here are the results I obtained (testing on PoseFix-H, test set):

* after pretraining on PoseFix-A:
   69.1 & 77.7 & 83.0 & 2.6 & 19.9 & 27.2 & 198 & 163 & 8.5
* after pretraining on PoseFix-A + finetuning on PoseFix-H:
   78.9 & 86.9 & 90.3 & 11.6 & 33.5 & 30.3 & 186 & 156 & 8.3

2. It the same dataset, but not the same splits (training and validation sets are used in the training phase; while the test set is used in final evaluation reports).

Additional question:
"GT"/"random"/"auto_posefix-A_cap0" give an idea of the results when substituting the generated text by:
* _GT_: the ground-truth (annotated) text
* _random_: a random human-written text from the dataset
* _auto_posefix-A_cap0:_ the first automatic (annotated) text of the sample of interest (the first one being supposedly closer to what a human would write, because of the use of some additional aggregation rules in the automatic pipeline)

**Comment by minjung98** [comment]
Thank you for your response, even if it was a bit delayed. 
To answer your question about which model I used: I used the` modret_distilbert_dataPFAftPFH` model, just like you mentioned. My goal was simply to reproduce the code, so aside from minor debugging issues like related to the CUDA environment, I ran your provided code as is and obtained the results. 
I also now understand the rest of your explanations below.(about dataset and additional question) After asking you the question, I looked more closely at the evaluation code and am glad to see my understanding was correct based on your reply.
It was due to my insufficient ability to read the code that I asked the question, but I am grateful and feel apologetic for receiving such a clear response from you.
